Foundation/Process.swift

```diff
@@ -232,8 +232,28 @@ open class Process: NSObject {
     }
 
     // These properties can only be set before a launch.
-    open var executableURL: URL?
-    open var currentDirectoryURL = URL(fileURLWithPath: FileManager.default.currentDirectoryPath, isDirectory: true)
+    private var _executable: URL?
+    open var executableURL: URL? {
+        get { _executable }
+        set {
+            guard let url = newValue, url.isFileURL else {
+                fatalError("must provide a launch path")
```

Comment:
Is `fatalError` correct?  Should this not be an exception of sorts?

Reply:
Darwin throws `NSException` so `fatalError` is the swift equivalent. Also it is a property so nothing else that could be done anyway.